Lib9c/Action/Sweep.cs

```diff
@@ -0,0 +1,263 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Bencodex.Types;
+using Libplanet;
+using Libplanet.Action;
+using Nekoyume.Battle;
+using Nekoyume.Extensions;
+using Nekoyume.Model;
+using Nekoyume.Model.Item;
+using Nekoyume.Model.Quest;
+using Nekoyume.Model.State;
+using Nekoyume.TableData;
+using static Lib9c.SerializeKeys;
+
+namespace Nekoyume.Action
+{
+    [Serializable]
+    [ActionType("sweep")]
+    public class Sweep : GameAction
+    {
+        public const int usableApStoneCount = 10;
+
+        public Address avatarAddress;
+        public int apStoneCount = 0;
+        public int worldId;
+        public int stageId;
+
+        protected override IImmutableDictionary<string, IValue> PlainValueInternal =>
+            new Dictionary<string, IValue>()
+            {
+                ["avatarAddress"] = avatarAddress.Serialize(),
+                ["apStoneCount"] = apStoneCount.Serialize(),
+                ["worldId"] = worldId.Serialize(),
+                ["stageId"] = stageId.Serialize(),
+            }.ToImmutableDictionary();
+
+        protected override void LoadPlainValueInternal(
+            IImmutableDictionary<string, IValue> plainValue)
+        {
+            avatarAddress = plainValue["avatarAddress"].ToAddress();
+            apStoneCount = plainValue["apStoneCount"].ToInteger();
+            worldId = plainValue["worldId"].ToInteger();
+            stageId = plainValue["stageId"].ToInteger();
+        }
+
+        public override IAccountStateDelta Execute(IActionContext context)
+        {
+            var states = context.PreviousStates;
+            var inventoryAddress = avatarAddress.Derive(LegacyInventoryKey);
+            var worldInformationAddress = avatarAddress.Derive(LegacyWorldInformationKey);
+            var questListAddress = avatarAddress.Derive(LegacyQuestListKey);
+            if (context.Rehearsal)
+            {
+                return states
+                    .SetState(inventoryAddress, MarkChanged)
+                    .SetState(worldInformationAddress, MarkChanged)
+                    .SetState(questListAddress, MarkChanged)
+                    .SetState(avatarAddress, MarkChanged)
+                    .SetState(context.Signer, MarkChanged);
+            }
+
+            var addressesHex = GetSignerAndOtherAddressesHex(context, avatarAddress);
+
+            if (!states.TryGetAvatarStateV2(context.Signer, avatarAddress, out var avatarState,
+                    out _))
```

Comment:
we can improve execution time by conditional serialize with `migrationRequired`

Reply:
I was Reflecting `migrationRequired`. But, thinking about it... worldInformation is not required for this action. 😅